msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/OAuthAuthorizationGrant.java

```diff
@@ -15,74 +13,54 @@
 
 class OAuthAuthorizationGrant extends AbstractMsalAuthorizationGrant {
```

Comment:
What is the point of the abstract class?

Reply:
It's existed since before I joined so I don't know the actual history of it. However, it appears to be copying the style of the `com.nimbusds.oauth2.sdk.AuthorizationGrant`, it specifies that the grant needs a `toParameters()` method and adds some MSAL-specific behavior (mainly just scopes?)

There are a few other classes that extend it, for example `DeviceCodeAuthorizationGrant` it has its own `toParameters()` that creates the parameters map in the same way that I did for the files in this PR: https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/dev/msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/DeviceCodeAuthorizationGrant.java